src/FFmpegReader.cpp

```diff
@@ -1315,8 +1319,23 @@ void FFmpegReader::ProcessVideoPacket(int64_t requested_frame) {
 		if (openshot::Settings::Instance()->HIGH_QUALITY_SCALING) {
 			scale_mode = SWS_LANCZOS;
 		}
-		SwsContext *img_convert_ctx = sws_getContext(info.width, info.height, AV_GET_CODEC_PIXEL_FORMAT(pStream, pCodecCtx), width,
-															  height, PIX_FMT_RGBA, scale_mode, NULL, NULL, NULL);
+		
+		// In case of 4:4:4 inputs, FFmpeg scaling optimization for fast_bilinear
+		// makes undesirable horizontal chroma simplifications.
+		// Do not reuse chroma for 2 pixels RGB/BGR in this case.
+		scale_mode |= SWS_FULL_CHR_H_INT;
+
+		SwsContext *img_convert_ctx = sws_getContext(info.width, info.height, pCodecCtx->pix_fmt, width,
+				height, PIX_FMT_RGBA, scale_mode, NULL, NULL, NULL);
+
+		// Scaling uses own numeration for color ranges Partial (mpeg, tv) = 0; Full (jpeg, pc) = 1;
+		// thus assuming that Partial color range was set.
+		int s_color_range = 0;
+		if (pCodecCtx->color_range == 2) s_color_range = Full_Range;
```

Comment:
Is `->color_range` supported in FFmpeg 2.x, 3.x, and 4.x? Just looks like a similar issue to pix_fmt, where FFmpeg refactors it somewhere else at some point.

Reply:
Available in all versions up to 1.0 (at least as storage) as far as I remember.